### ISSUE TITLE
Access error body

### ISSUE THREAD

**Issue by andys8** [iarthstar/purescript-axios#4]
**Is your feature request related to a problem? Please describe.**

The request returns type [`Error`](https://pursuit.purescript.org/packages/purescript-exceptions/4.0.0/docs/Effect.Exception#t:Error). It does contain a message like: "Request failed with status code 401". It looks like there is no way to access the http body which contains additional information.

**Describe the solution you'd like**
Provide a way to access the response body in error cases.

**Describe alternatives you've considered**
It might be already possible and I'm missing something.

**Comment by iarthstar** [comment]
Hey @andys8, I'll get back to this asap...

**Comment by andys8** [comment]
Hey @iarthstar , at some point in the future, I think I'm going to need this to be able to use the library. If possible you could start sketching your idea of how api and implementation should look like. Maybe I can help implementing it.

**Comment by iarthstar** [comment]
Yeah, I'm thinking we need to change `Response a` -> `Either HttpBody a`

Do let me know your thoughts on the same or any other improvements...

Also what all keys and hierarchy you expect from http body... if possible do share the type...

**Comment by andys8** [comment]
For comparison: Elm's http library has a response type and metadata

``` 
type Response body
    = BadUrl_ String
    | Timeout_
    | NetworkError_
    | BadStatus_ Metadata body
    | GoodStatus_ Metadata body

type alias Metadata =
    { url : String
    , statusCode : Int
    , statusText : String
    , headers : Dict String String
    }
```

https://package.elm-lang.org/packages/elm/http/2.0.0/Http#Response

So one can either go the full blown way, or come up with a simpler response type that could work in these cases.

**Comment by iarthstar** [comment]
The Elm's http response type can solve pretty much all the current and future requirements for Response, so I think I'll look into this and start the implementation asap...

**Comment by andys8** [comment]
For completeness: Elm has also an `Error` type that is missing some information.

```elm
type Error
    = BadUrl String
    | Timeout
    | NetworkError
    | BadStatus Int
    | BadBody String
```

https://package.elm-lang.org/packages/elm/http/2.0.0/Http#Error

I personally am in favour of a signature where the good case is nice to use. So maybe offer two ways to fire a request. And in both cases I would go with `Either` instead of a large sum type for the good and bad case.

```
request :: .. -> Either RequestError a
request' :: .. -> Either RequestError (ResponseWithMetadata a)
